### PR TITLE
Next

### PR DESCRIPTION
**Pull request by Nikratio**
Nothing new in terms of features here, this is all preparation for other upcoming work.

### CODE REVIEW

File: src/s3ql/backends/gs.py [hunk 1]
```diff
@@ -102,6 +101,56 @@ class AccessTokenExpired(Exception):
     '''
 
 
+def _parse_error_response(resp, body: bytes):
```
[the remark this answers is not in the document]
Comment (Nikratio) [reply src/s3ql/backends/gs.py:104]: This is refactoring of existing code, just moved from one place in the file to another.

File: src/s3ql/backends/gs.py [hunk 2]
```diff
@@ -102,6 +101,56 @@ class AccessTokenExpired(Exception):
...
+    try:
+        body_text = body.decode(charset)
+    except UnicodeDecodeError as exc:
+        log.warning('Unable to decode JSON response as Unicode (%s) - this is odd.', str(exc))
+        raise ServerResponseError(resp, error=str(exc))
+
+    try:
+        resp_json = json.loads(body_text)
+    except json.JSONDecodeError as exc:
+        log.warning('Unable to decode JSON response (%s) - this is odd.', str(exc))
```
[the remark this answers is not in the document]
Comment (Nikratio) [reply src/s3ql/backends/gs.py:148]: Yes, arguably. But I'd like to keep such cleanups/improvements from refactoring.

[the remark this answers is not in the document]
Comment (IsaacG) [reply src/s3ql/backends/gs.py:148]: It'd be helpful to know what sort of things you are/aren't looking for in a review 😄

[the remark this answers is not in the document]
Comment (Nikratio) [reply src/s3ql/backends/gs.py:148]: Ideally any subset of the following:
 
- Bugs in code (doesn't do what it's probably supposed to do)
- Bugs in intent (it does what it seemingly should do, but that's not a good idea)
- Code that's difficult to understand
- Documentation that's incorrect or difficult to understand

File: src/s3ql/backends/gs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nikratio) [reply src/s3ql/backends/gs.py]: (fixed)

**Pull request merged**